test/e2e/overrides_test.go

```diff
@@ -140,13 +140,15 @@ var _ = Describe("Overrides test", Ordered, func() {
 		})
 	})
 
-	AfterAll(func() {
+	AfterEach(func() {
 		By("Reset cert-manager state")
 		err := resetCertManagerState(context.Background(), certmanageroperatorclient, loader)
 		Expect(err).NotTo(HaveOccurred())
+	})
```

Comment:
This change seems to be an overkill. The`BeforeEach` section already has the call to the `resetCertManagerState`. Adding the same in `AfterEach` section will essentially increase the calls to the api-server and the reconciles.

Reply:
Yeah I understand, but without the `resetCertManagerState` in both BeforeEach and AfterEach, what ends up happening is that thre is unclean state of the operator by the end of some test, which affects the other tests and fails out.
eg. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cert-manager-operator/140/pull-ci-openshift-cert-manager-operator-master-e2e-operator/1694727559633178624, https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cert-manager-operator/140/pull-ci-openshift-cert-manager-operator-master-alpha-e2e-operator/1694727559549292544 it's reproducible across runs; adding this resolves it.

Reply:
That doesn't seem to have resolved the issue I think.
Webhook still goes down and test flakes altogether.
Recent run reports webhook failure: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cert-manager-operator/140/pull-ci-openshift-cert-manager-operator-master-e2e-operator/1694970785124323328.

Reply:
> Yeah I understand, but without the `resetCertManagerState` in both BeforeEach and AfterEach, what ends up happening is that thre is unclean state of the operator by the end of some test, which affects the other tests and fails out. eg. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cert-manager-operator/140/pull-ci-openshift-cert-manager-operator-master-e2e-operator/1694727559633178624, https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cert-manager-operator/140/pull-ci-openshift-cert-manager-operator-master-alpha-e2e-operator/1694727559549292544 (reconcile failures in: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cert-manager-operator/140/pull-ci-openshift-cert-manager-operator-master-e2e-operator/1694727559633178624/artifacts/e2e-operator/gather-extra/artifacts/pods/cert-manager-operator_cert-manager-operator-controller-manager-f6b7fd94d-7vdsw_cert-manager-operator.log) it's reproducible across runs; adding this resolves it, and tests pass.

The Override tests shouldn't effect the Certificates test as both the ci runs the Certificates test is run before the Overrides tests. So this seems unrelated to the additional call to the `resetCertManagerState`.

Reply:
Yeah, I checked those logs. certificates_test was executed before overrides_test and all the steps in overrides_test did pass.
However, I found that TestContainerOverrides fn (the old non-ginko test) is always being executed before TestAll (the new ginkgo suite) and there's possibility for that to have caused the regressive behaviour not cleaning up it's own overrides.

I guess this isn't cleaning up properly,
```go

func TestContainerOverrides(t *testing.T) {
         // ...

        // reset operator spec
	operator, err = certmanageroperatorclient.OperatorV1alpha1().CertManagers().Get(ctx, "cluster", metav1.GetOptions{})
	require.NoError(t, err)

	updatedOperator = operator.DeepCopy()
	updatedOperator.Spec = v1alpha1.CertManagerSpec{
		OperatorSpec: opv1.OperatorSpec{
			ManagementState: opv1.Managed,
		},
	}
	_, err = certmanageroperatorclient.OperatorV1alpha1().CertManagers().Update(ctx, updatedOperator, metav1.UpdateOptions{})
	require.NoError(t, err)
}
```

Reply:
Something is still wrong (maybe with TestContainerOverrides func), as with the addOverrides/resetCertManager removed from certificates_test also resulted in the same "webhook unavailable error".

Reply:
Replaced the delete with `resetCertManagerState` func call now. This should resolve it!
Let's wait for e2e to pass one final time.